src/algo/camera-age/l500/camera-age.cpp

```diff
@@ -11,22 +10,39 @@ namespace algo {
 namespace camera_age {
 namespace l500 {
 
-unsigned manufacture_time::get_manufacture_year() const
+work_week::work_week( std::time_t & t )
+{
+    auto time = std::localtime( t );
+    int year = time->tm_year + 1900;  // The tm_year field contains the number of years
+                                      // since 1900, we add 1900 to get current year
+    int ww = time->tm_yday / 7;       // The tm_yday field contains the number of days sine 1st
+                                      // of January, we divide by 7 to get current work week
+    ww++;  // We add 1 because work weeks start at 1 and not 0 (ex. 1st of January will return
+           // ww 0, but it ww 1)
+    return work_week( year, ww );
+}
+
+unsigned work_week::get_year() const
 {
     return man_year;
 }
 
-unsigned manufacture_time::get_manufacture_work_week() const
+unsigned work_week::get_work_week() const
 {
     return man_ww;
 }
 
+unsigned work_week::operator-( const work_week & ww ) const
+{
+    return ( ( this->get_year() - ww.get_year() ) * 52 )  // There are 52 weeks in a year
```

Comment:
Some years have a 53rd week...
I wonder if we need to worry about ww1-ww53? Will this work?

Reply:
when I spoke to Dubi he said this would be fine

Reply:
This, BTW, needs to be covered in a unit-test